WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/DefaultProductFormTableViewModelTests.swift

```diff
@@ -265,6 +265,106 @@ final class DefaultProductFormTableViewModelTests: XCTestCase {
         XCTAssertEqual(subscriptionViewModel?.details, expectedDetails)
     }
 
+    // MARK: Subscription free trial
+
+    func test_subscription_free_trial_row_returns_expected_details_with_singular_format() throws {
+        // Given
+        let subscription = ProductSubscription.fake().copy(trialLength: "1", trialPeriod: .month)
+        let product = Product.fake().copy(productTypeKey: ProductType.subscription.rawValue, subscription: subscription)
+        let model = EditableProductModel(product: product)
+        let actionsFactory = ProductFormActionsFactory(product: model, formType: .edit, editingSubscriptionEnabled: true)
+        let currencyFormatter = CurrencyFormatter(currencySettings: CurrencySettings())
+
+        // When
+        let tableViewModel = DefaultProductFormTableViewModel(product: model,
+                                                              actionsFactory: actionsFactory,
+                                                              currency: "",
+                                                              currencyFormatter: currencyFormatter,
+                                                              isDescriptionAIEnabled: true)
+
+        // Then
+        guard case let .settings(rows) = tableViewModel.sections[1] else {
+            XCTFail("Unexpected section at index 1: \(tableViewModel.sections)")
+            return
+        }
+        var subscriptionFreeTrialViewModel: ProductFormSection.SettingsRow.ViewModel?
+        for row in rows {
+            if case let .subscriptionFreeTrial(viewModel, _) = row {
+                subscriptionFreeTrialViewModel = viewModel
+                break
+            }
+        }
+
+        let viewModel = try XCTUnwrap(subscriptionFreeTrialViewModel)
+        XCTAssertEqual(viewModel.details, "1 month")
```

Comment:
👀 This works because our CI runs on `en` locale, but I believe it's worth being mindful when testing localized labels. It's best practice to compare with an `NSLocalizedString` value.

Reply:
Good point. Done in 710c86227d6def489988dd2a48acace50a5b30c5